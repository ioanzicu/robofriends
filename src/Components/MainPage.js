import React, { useEffect } from 'react'
import CardList from '../Components/CardList'
import SearchBox from '../Components/SearchBox'
import Scroll from '../Components/Scroll'
import ErrorBoundry from '../Components/ErrorBoundry'
import { MemoizedHeader } from '../Components/Header'

const MainPage = ({
  searchField,
  onSearchChange,
  robots,
  isPending,
  onRequestRobots
}) => {
  useEffect(() => {
    onRequestRobots()
  }, [onRequestRobots])

  const filterRobots = robots.filter(robot =>
    robot.name.toLowerCase().includes(searchField.toLowerCase())
  )

  const title = isPending ? 'Loading...' : 'Robo Friends'

  return (
    <div className='tc'>
      <MemoizedHeader title={title} />
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        <ErrorBoundry>
          <CardList robots={filterRobots} />
        </ErrorBoundry>
      </Scroll>
    </div>
  )
}

export default MainPage
