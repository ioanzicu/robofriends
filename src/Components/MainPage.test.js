import React from 'react'
import { shallow } from 'enzyme'
import MainPage from './MainPage'

let wrapper
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps} />)
})

it('renders MainPage without crashing', () => {
  expect(wrapper.debug()).toMatchSnapshot()
})
