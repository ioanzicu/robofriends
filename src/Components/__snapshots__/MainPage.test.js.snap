// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
"<div className=\\"tc\\">
  <Memo() title=\\"Robo Friends\\" />
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundry>
      <CardList robots={{...}} />
    </ErrorBoundry>
  </Scroll>
</div>"
`;
